code/modules/mob/living/carbon/human/species.dm

```diff
@@ -750,107 +752,115 @@
-				playsound(H.loc, M.dna.species.attack_sound, 25, 1, -1)
-			else
-				playsound(H.loc, 'sound/weapons/punch1.ogg', 25, 1, -1)
+				var/obj/item/organ/limb/affecting = H.get_organ(ran_zone(M.zone_sel.selecting))
+				var/armor_block = H.run_armor_check(affecting, "melee")
 
+				if(M.dna)
+					playsound(H.loc, M.dna.species.attack_sound, 25, 1, -1)
+				else
+					playsound(H.loc, 'sound/weapons/punch1.ogg', 25, 1, -1)
 
-			H.visible_message("<span class='danger'>[M] has [atk_verb]ed [H]!</span>", \
-							"<span class='userdanger'>[M] has [atk_verb]ed [H]!</span>")
 
-			H.apply_damage(damage, BRUTE, affecting, armor_block)
-			if((H.stat != DEAD) && damage >= 9)
-				H.visible_message("<span class='danger'>[M] has weakened [H]!</span>", \
-								"<span class='userdanger'>[M] has weakened [H]!</span>")
-				H.apply_effect(4, WEAKEN, armor_block)
-				H.forcesay(hit_appends)
-			else if(H.lying)
-				H.forcesay(hit_appends)
+				H.visible_message("<span class='danger'>[M] has [atk_verb]ed [H]!</span>", \
+								"<span class='userdanger'>[M] has [atk_verb]ed [H]!</span>")
 
+				H.apply_damage(damage, BRUTE, affecting, armor_block)
+				if((H.stat != DEAD) && damage >= 9)
+					H.visible_message("<span class='danger'>[M] has weakened [H]!</span>", \
+									"<span class='userdanger'>[M] has weakened [H]!</span>")
+					H.apply_effect(4, WEAKEN, armor_block)
+					H.forcesay(hit_appends)
+				else if(H.lying)
+					H.forcesay(hit_appends)
 		if("disarm")
-			M.do_attack_animation(H)
-			add_logs(M, H, "disarmed")
-
-			if(H.w_uniform)
-				H.w_uniform.add_fingerprint(M)
-			var/obj/item/organ/limb/affecting = H.get_organ(ran_zone(M.zone_sel.selecting))
-			var/randn = rand(1, 100)
-			if(randn <= 25)
-				H.apply_effect(2, WEAKEN, H.run_armor_check(affecting, "melee"))
-				playsound(H, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
-				H.visible_message("<span class='danger'>[M] has pushed [H]!</span>",
-								"<span class='userdanger'>[M] has pushed [H]!</span>")
-				H.forcesay(hit_appends)
-				return
-
-			var/talked = 0	// BubbleWrap
-
-			if(randn <= 60)
-				//BubbleWrap: Disarming breaks a pull
-				if(H.pulling)
-					H.visible_message("<span class='warning'>[M] has broken [H]'s grip on [H.pulling]!</span>")
-					talked = 1
-					H.stop_pulling()
-
-				//BubbleWrap: Disarming also breaks a grab - this will also stop someone being choked, won't it?
-				if(istype(H.l_hand, /obj/item/weapon/grab))
-					var/obj/item/weapon/grab/lgrab = H.l_hand
-					if(lgrab.affecting)
-						H.visible_message("<span class='warning'>[M] has broken [H]'s grip on [lgrab.affecting]!</span>")
-						talked = 1
-					spawn(1)
-						qdel(lgrab)
-				if(istype(H.r_hand, /obj/item/weapon/grab))
-					var/obj/item/weapon/grab/rgrab = H.r_hand
-					if(rgrab.affecting)
-						H.visible_message("<span class='warning'>[M] has broken [H]'s grip on [rgrab.affecting]!</span>")
+			if(attacker_style && attacker_style.disarm_act(M,H))
+				return 1
+			else:
```

Comment:
iirc @AnturK is working on other shit in other languages.
I know python uses colons and such, so it's probably just force of habit.